aesara/tensor/type.py

```diff
@@ -332,55 +334,90 @@ def values_eq_approx(
     ):
         return values_eq_approx(a, b, allow_remove_inf, allow_remove_nan, rtol, atol)
 
-    def __hash__(self):
-        return hash((type(self), self.dtype, self.broadcastable))
-
-    ndim = property(lambda self: len(self.broadcastable), doc="number of dimensions")
-    """
-    Number of dimensions.
+    def __eq__(self, other):
+        if type(self) != type(other):
+            return NotImplemented
 
-    This read-only property is the preferred way to get the number of
-    dimensions of a `TensorType`.
+        return other.dtype == self.dtype and other.shape == self.shape
 
-    """
-
-    def make_variable(self, name=None):
-        """
-        Return a `TensorVariable` of this type.
+    def __hash__(self):
+        return hash((type(self), self.dtype, self.shape))
 
-        Parameters
-        ----------
-        name : str
-            A pretty name to identify this `Variable` when printing and
-            debugging
+    @property
+    def broadcastable(self):
+        """A boolean tuple indicating which dimensions have a shape equal to one."""
+        return tuple(s == 1 for s in self.shape)
```

Comment:
> Is there a (way to) warn about the broadcastable deprecation before removing it completely?

@ricardoV94 It's still there except that it's now derived from shape. I guess we can put in a warning here if we intend to remove it.

Reply:
Thanks, I was missing it!

Reply:
Oh, I hadn't planned to remove that.  We will need to change a lot more things after this PR is merged, but those are related to using more static shape information, instead of the `broadcastable` values that we've been using.